packages/contracts-core/contracts/test/utils/NomadCore.sol

```diff
@@ -0,0 +1,340 @@
+
+        governanceRouterImplementation = new GovernanceRouter(
+            localDomain,
+            recoveryTimelock
+        );
+        governanceRouterController = new UpgradeBeaconController();
+        governanceRouterBeacon = new UpgradeBeacon(
+            address(governanceRouterImplementation),
+            address(governanceRouterController)
+        );
+        data = abi.encodeWithSignature(
+            "initialize(address,address)",
+            address(xAppConnectionManager),
+            recoveryManager
+        );
+        governanceRouterProxy = new UpgradeBeaconProxy(
+            address(governanceRouterBeacon),
+            data
+        );
+        // Create a GovernaceRouter object so that we can call the Proxie's functions
+        // as if we would call the contract itself.
+        governanceRouter = GovernanceRouter(address(governanceRouterProxy));
+
+        governanceRouters.push(governanceRouter);
+        governanceRouterProxies.push(governanceRouterProxy);
+        governanceRouterImplementations.push(governanceRouterImplementation);
+        governanceRouterBeacons.push(governanceRouterBeacon);
+        governanceRouterControllers.push(governanceRouterController);
+
+        domainToIndex[localDomain] = homes.length - 1;
+    }
+
+    function createLocalReplicaForRemoteDomain(
+        uint32 localDomain,
+        uint32 remoteDomain
+    ) public {
+        bytes memory data;
+        replicaImplementation = new Replica(localDomain);
+        replicaController = new UpgradeBeaconController();
+        replicaBeacon = new UpgradeBeacon(
+            address(replicaImplementation),
+            address(replicaController)
+        );
+        data = abi.encodeWithSignature(
+            "initialize(uint32,address,bytes32,uint256)",
+            remoteDomain,
+            updater,
+            committedRoot,
+            optimisticSeconds
+        );
+        replicaProxy = new UpgradeBeaconProxy(address(replicaBeacon), data);
+        replica = Replica(address(replicaProxy));
+
+        replicas[localDomain][remoteDomain] = replica;
+        replicaProxies[localDomain][remoteDomain] = replicaProxy;
+        replicaImplementations[localDomain][
+            remoteDomain
+        ] = replicaImplementation;
+        replicaBeacons[localDomain][remoteDomain] = replicaBeacon;
+        replicaControllers[localDomain][remoteDomain] = replicaController;
+        uint256 index = domainToIndex[localDomain];
+        xAppConnectionManagers[index].ownerEnrollReplica(
+            address(replica),
+            remoteDomain
+        );
+        replicaControllers[localDomain][remoteDomain].transferOwnership(
+            address(governanceRouters[domainToIndex[localDomain]])
+        );
+    }
+
+    function relinquishCoreControl(uint32 localDomain) public {
+        uint256 index = domainToIndex[localDomain];
+        governanceRouter = governanceRouters[index];
```

Comment:
TODO: this setup doesn't have a concept of governorDomain. it would be closer to the true setup to have a governorDomain and non-governerDomain. need to call `transferGovernor` on both domains

Reply:
also need to add `governor` as a system actor (next to `recoveryManager`)